src/argilla/server/models/questions.py

```diff
@@ -52,17 +55,41 @@ class RatingQuestionSettingsOption(BaseModel):
     value: int
 
 
-class RatingQuestionSettings(BaseQuestionSettings):
-    type: Literal[QuestionType.rating]
-    options: List[RatingQuestionSettingsOption]
+T = TypeVar("T")
 
+
+class ValidOptionCheckerMixin(BaseQuestionSettings, Generic[T]):
     @property
-    def option_values(self) -> List[int]:
+    def option_values(self) -> List[T]:
         return [option.value for option in self.options]
 
     def check_response(self, response: ResponseValue):
         if response.value not in self.option_values:
             raise ValueError(f"{response.value!r} is not a valid option.\nValid options are: {self.option_values!r}")
 
 
-QuestionSettings = Annotated[Union[TextQuestionSettings, RatingQuestionSettings], Field(..., discriminator="type")]
+class RatingQuestionSettings(ValidOptionCheckerMixin[int], BaseQuestionSettings):
+    type: Literal[QuestionType.rating]
+    options: List[RatingQuestionSettingsOption]
+
+
+class LabelSelectionQuestionSettingsOption(BaseModel):
+    value: str
+    text: str
+    description: Optional[str] = None
```

Comment:
Maybe make sense having some length limits here

Reply:
I've put my comment but I didn't push it. We could align to the [questions and fields limits](https://github.com/argilla-io/argilla/pull/3005/files#diff-154a4fe702792a783f557b4e95846e0f9f58dc1c430df9afd0e9d2206f2ef918R35):

* `name`/`value`: 200
* `title`/`text`: 500
* `description`: 1000

My concern about lower values is to be too much restrictive.